aten/src/ATen/native/SoftMax.cpp

```diff
@@ -141,6 +197,28 @@ Tensor softmax_cpu(const Tensor& input_, const int64_t dim_, const bool half_to_
   return output;
 }
 
+Tensor masked_softmax_cpu(const Tensor& input_, const Tensor& mask_, const int64_t dim_, const bool half_to_float) {
```

Comment:
You could add two more checks here.
One to verify that both mask and input are actually on CPU and another to check the dtypes and optionally mask type conversion.